core/payment/src/processor.rs

```diff
@@ -655,25 +655,24 @@ impl PaymentProcessor {
 
         tokio::task::spawn_local(
             async move {
-                let db_executor = db_executor
-                    .timeout_lock(DB_LOCK_TIMEOUT, "notify payment 2")
-                    .await?;
-
-                let payment_dao: PaymentDao = db_executor.as_dao();
-                let sync_dao: SyncNotifsDao = db_executor.as_dao();
-
                 let send_result =
                     Self::send_to_gsb(payer_id, payee_id, msg_with_bytes.clone()).await;
 
-                let mark_sent = if send_result.is_ok() {
-                    payment_dao
-                        .add_signature(
-                            payment_id.clone(),
-                            msg_with_bytes.signature.clone(),
-                            msg_with_bytes.signed_bytes.clone(),
-                        )
-                        .await
-                        .is_ok()
+                // Mutablility increases code readability - otherwise we'd have to take the lock
+                // and create payment_dao in every branch of the if-else below because we have to
+                // ensure that it is taken after the remote call performed only in some of the
+                // branches.
```

Comment:
Note there is a nice trick with setting multiple values at once conditionally (as blocks of code return values in Rust) - see https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=76aeaf7adb0ca4f079fbd2e789c9d0b3 for an example